lib/signatures.ts

```diff
@@ -1,25 +1,34 @@
 import { promises } from 'fs';
-import { SignatureHashAlgorithm, SignatureResult } from './types';
+import { SignatureFormat, SignatureResult } from './types';
 import { getDubHashSignature } from './dubhash';
 import { getUhashSignature } from './uhash';
 import pMap = require('p-map');
 import { join } from 'path';
 
 const { readFile } = promises;
 
+type Signer = (path: string, content: Buffer) => SignatureResult | null;
+interface SignatureFormatMap {
+  [SignatureFormat: string]: Signer;
+}
+
+const algorithms: SignatureFormatMap = {
+  dubhash: getDubHashSignature,
+  uhash: getUhashSignature,
+};
+
 export async function getSignaturesByAlgorithm(
   baseDir: string,
   filePaths: string[],
-  hashType: SignatureHashAlgorithm = 'dubhash',
+  hashType: SignatureFormat = 'dubhash',
 ): Promise<(SignatureResult | null)[]> {
-  if (hashType !== 'dubhash' && hashType !== 'uhash') {
+  const getSignature = algorithms[hashType];
+
+  if (getSignature === undefined) {
     throw new Error(`Unsupported hashType ${hashType}`);
   }
 
-  const getSignature =
-    hashType === 'uhash' ? getUhashSignature : getDubHashSignature;
-
-  const mapper = async (path: string) => {
+  const pathToSignature = async (path: string) => {
```

Comment:
This name seems off. It should sound like a function, but this sounds more like a "noun".

Reply:
@robguinness-snyk: As you know from our DM last week I struggled with the name, all names I considered seemed off.

Just before I decided to go with this one I also considered toSignature, since I'm used to having a struct and "member" functions named like that. But since it's not a struct I thought I needed "prefix" it to something more describing.

To be honest, it still feels strange, since it's not a conversion from one format to another. In my opinion names beginning with "to" and "from" is perfect for some kind of "constructors" but that's not really the case here. 

`sign`, `getSignature` was considered as well. The first one would work, since it's a verb, but I thought it might not be specific enough for some developers, and the latter one was already taken. To make it more specific it could be "signFile", hmm.

Also I don't really like the `get`-prefix in this case, since it's something that would be calculated. "get" sounds like classical getter and setter, which is used in OOP, and that's not really the case either.

Maybe calc, compute or hash could be used, but I'm not sure.

Reply:
I still don't like the word get, it's not just that it's used by the OOP with getters and setters, it's also indicating that we are fetching something that already has been precalculated or an instance that exists, but that's just my opinion and I'm always open to stick with it.

Reply:
maybe generateSignatureFromPath